src/yaml.rs

```diff
@@ -0,0 +1,95 @@
+//! Utility functions for processing data in the YAML file format
+use serde::ser;
+
+use crate::error::{Error, OperatorResult};
+
+/// A YAML document type
+///
+/// For a detailled description, see the
+/// [YAML specification](https://yaml.org/spec/1.2.2/#rule-l-any-document).
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum DocumentType {
+    DirectiveDocument,
+    ExplicitDocument,
+    BareDocument,
+}
+
+/// Serializes the given data structure as an explicit YAML document.
+///
+/// # Errors
+///
+/// Serialization can fail if `T`'s implementation of `Serialize` decides to return an error.
+///
+/// An [`Error::UnsupportedYamlDocumentError`] is returned if the used `serde_yaml` version
+/// generates a directive document.
+pub fn to_explicit_document_string<T>(value: &T) -> OperatorResult<String>
+where
+    T: ?Sized + ser::Serialize,
+{
+    // The returned document type depends on the serde_yaml version.
+    let document = serde_yaml::to_string(value)?;
```

Comment:
So, you would abort if the returned YAML document type is not the expected one according to the `serde_yaml` version? For instance:

```
assert!(!document.starts_with("---") && !document.starts_with('%'));
```

I did it this way because in my opinion the "contract" with `serde_yaml` is not defined by its version and changelog but by its documentation. The `to_string` function returns "a String of YAML" according to its documentation and that has not changed over the versions. So I can expect a valid YAML document of any document type and therefore I have to check the type and do the necessary conversion. Unfortunately the directive document case is too complex that I am willing to implement it.

But I see that the reduced complexity also has its value. So I would change it if my comment did not convince you.

Reply:
I'd put that check into a `#[test]` so that we detect it when upgrading.

The change is listed as a breaking in `serde_yaml`'s [changelog](https://github.com/dtolnay/serde-yaml/releases/tag/0.9.0), implying that any future change would also have to be semver-major (and thus has to be explicitly approved and tested for operator-rs).

Reply:
Document type check removed. If `serde_yaml::to_string` does not return the expected type then the already existing unit test fails (tested with `serde_yaml:0.8`):

```
thread 'yaml::tests::value_can_be_serialized_to_an_explicit_document_string' panicked at 'assertion failed: `(left == right)`
  left: `"---\nkey: value\n"`,
 right: `"---\n---\nkey: value\n"`', src/yaml.rs:37:9
```